src/python/zquantum/core/wip/estimators/estimation_interface.py

```diff
@@ -26,21 +26,19 @@ class EstimationTask:
     number_of_shots: int
 
 
-class EstimationTaskTransformer(Protocol):
+class EstimationPreprocessor(Protocol):
     """Protocol defining function which transforms a list of EstimationTasks
     into another list of EstimationTasks.
     """
 
-    def __call__(
-        self, estimation_problems: List[EstimationTask], **kwargs
-    ) -> List[EstimationTask]:
+    def __call__(self, estimation_tasks: List[EstimationTask]) -> List[EstimationTask]:
```

Comment:
No kwargs in the end?

Reply:
Nope - I thought we agreed we would use partials to avoid having kwargs

Reply:
I thought we agreed on the opposite 😆 
Both are fine with me ;)